.github/workflows/job_PreCommit_Kotlin_Examples.yml

```diff
@@ -0,0 +1,94 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PreCommit Kotlin Examples
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths:
+    - 'model/**'
+    - 'sdks/java/**'
+    - 'runners/flink/**'
+    - 'runners/spark/**'
+    - 'runners/direct-java/**'
+    - 'examples/kotlin/**'
+    - 'release/**'
+    - '.github/workflows/job_PreCommit_Kotlin_Examples.yml'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    paths:
+      - 'model/**'
+      - 'sdks/java/**'
+      - 'runners/flink/**'
+      - 'runners/spark/**'
+      - 'runners/direct-java/**'
+      - 'examples/kotlin/**'
+      - 'release/**'
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '* */6 * * *'
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+jobs:
+  KotlinExamplesPreCommit:
```

Comment:
We should use the same job name as Jenkins to keep it consistent. Please rename to `job_PreCommit_Kotlin_Examples`